### PR TITLE
Add DisplayLink sleep support embedded in run file

### PR DESCRIPTION
**Pull request by ssaavedra**

### CODE REVIEW

File: displaylink-sleep.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# Copyright (c) 2015 DisplayLink (UK) Ltd.
```
Comment (hadess) [inline displaylink-sleep.sh]: Where is that file from? Is there any document explaining the messages you're sending to those sockets?

**Comment by hadess** [comment]
Looks good to me, apart from the lack of docs and explanations (the helper has a copyright but no license, we don't know where it's from or what it does).

**Comment by ssaavedra** [comment]
The file is the result of function `add_pm_script` in displaylink-installer.sh, one of the files that gets decompressed after running `./displaylink-driver-1.1.62.run --keep --noexec`. It is extracted in this same way in ArchLinux's AUR for the same reasons as here: repackaging for the package manager.

The extracted final text is the same on systemd-enabled distributions, so this should still fall inside the "no modification" condition in the license, at least, seeing that although they do not officially support other distros, they seem to be happy with other distros repackaging the driver.

In any case, you are right that this should have docs and explanations. If you can have any input there, it would be great. I really dislike the fact that DisplayLinkDriver is not open-source in the first place. We might even be able to fix the annoying blinking-cursor bug.

**Comment by hadess** [comment]
If the file is a modification of what's in the self-extracting archive, I'd rather we got to it from there, rather than adding new code. So we'd call a helper in the .spec:
```
#/bin/sh

COREDIR=path/to/buildroot
source path/to/displaylink-installer.sh
add_pm_script "systemd"
```

That way, we're sure that the helper script is the same as in the vendor sources.

**Comment by ssaavedra** [comment]
Wow, didn't think of that. I'll patch the PR in a moment.

2016-05-30 18:27 GMT+02:00 hadess <notifications@github.com>:

**Comment by ssaavedra** [comment]
It was a bit more complicated than I expected, due to the fact that it is a script, so I trimmed the rest of it, so that it wouldn't fail for not being UID=0 or doing other stuff than what it was supposed to do.

**Comment by hadess** [comment]
Looked great, thanks for the patch.

**Pull request merged**